service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java

```diff
@@ -93,6 +109,66 @@ public void recordStewardship(
   }
 
   @Override
+  public void updateSite(JWTData userData, int siteId, UpdateSiteRequest updateSiteRequest) {
+    checkSiteExists(siteId);
+
+    SiteEntriesRecord record = db.newRecord(SITE_ENTRIES);
+
+    record.setUserId(userData.getUserId());
+    record.setSiteId(siteId);
+    record.setUpdatedAt(new Timestamp(System.currentTimeMillis()));
+    record.setTreePresent(updateSiteRequest.isTree_present());
+    record.setStatus(updateSiteRequest.getStatus());
+    record.setGenus(updateSiteRequest.getGenus());
+    record.setSpecies(updateSiteRequest.getSpecies());
+    record.setCommonName(updateSiteRequest.getCommon_name());
+    record.setConfidence(updateSiteRequest.getConfidence());
+    record.setDiameter(updateSiteRequest.getDiameter());
+    record.setCircumference(updateSiteRequest.getCircumference());
+    record.setMultistem(updateSiteRequest.isMultistem());
+    record.setCoverage(updateSiteRequest.getCoverage());
+    record.setPruning(updateSiteRequest.getPruning());
+    record.setCondition(updateSiteRequest.getCondition());
+    record.setDiscoloring(updateSiteRequest.isDiscoloring());
+    record.setLeaning(updateSiteRequest.isLeaning());
+    record.setConstrictingGate(updateSiteRequest.isConstricting_gate());
+    record.setWounds(updateSiteRequest.isWounds());
+    record.setPooling(updateSiteRequest.isPooling());
+    record.setStakesWithWires(updateSiteRequest.isStakes_with_wires());
+    record.setStakesWithoutWires(updateSiteRequest.isStakes_without_wires());
+    record.setLight(updateSiteRequest.isLight());
+    record.setBicycle(updateSiteRequest.isBicycle());
+    record.setBagEmpty(updateSiteRequest.isBag_empty());
+    record.setBagFilled(updateSiteRequest.isBag_filled());
+    record.setTape(updateSiteRequest.isTape());
+    record.setSuckerGrowth(updateSiteRequest.isSucker_growth());
+    record.setSiteType(updateSiteRequest.getSite_type());
+    record.setSidewalkWidth(updateSiteRequest.getSidewalk_width());
+    record.setSiteWidth(updateSiteRequest.getSite_width());
+    record.setSiteLength(updateSiteRequest.getSite_length());
+    record.setMaterial(updateSiteRequest.getMaterial());
+    record.setRaisedBed(updateSiteRequest.isRaised_bed());
+    record.setFence(updateSiteRequest.isFence());
+    record.setTrash(updateSiteRequest.isTrash());
+    record.setWires(updateSiteRequest.isWires());
+    record.setGrate(updateSiteRequest.isGrate());
+    record.setStump(updateSiteRequest.isStump());
+    record.setTreeNotes(updateSiteRequest.getTree_notes());
+    record.setSiteNotes(updateSiteRequest.getSite_notes());
+
+    record.store();
+  }
+
+  @Override
+  public void deleteSite(JWTData userData, int siteId) {
+    isAdminCheck(userData.getPrivilegeLevel());
```

Comment:
⛏️ IMO we shouldnt have a void method `isAdminCheck` whose only purpose is to throw. I would rather have that logic as a boolean and then throw in the handler itself

Reply:
We follow this same pattern for teams and registrations. I can make a ticket to change all of them